### ISSUE TITLE
Requesting some enlightenment on test.csv

### ISSUE THREAD

**Issue by shakjm** [MASILab/tumor-cifar#1]
Hi Mr. Gao,

Thank you so much for sharing your work with us. It is indeed an interesting read! May I know if there is a test.csv available? Because it seems that it has been removed and we have no access to it. Is it self generated?

Also, may I know if these csv files are generated at random? How did you produce these test.csv and train.csv files?

**Comment by shakjm** [comment]
Also, Mr. Gao, I've tried using your previous github version of train.csv and test.csv. May I know why the training and AUC stays only at 0.5 after training for so many iterations?

**Comment by RiqiangGao** [comment]
> Also, Mr. Gao, I've tried using your previous github version of train.csv and test.csv. May I know why the training and AUC stays only at 0.5 after training for so many iterations?

Hi, Shakjm, thanks for your interest. As reported in the paper, in the v2 of tumor-cifar, there is no clear difference between the pos / neg if only consider the image. So, the classical CNN / LSTM can only achieve ~0.5. As for the *.csv files, they can be generated using the data_submit.py, examples in data_submit.ipynb. Note that, there is randomness in simulating the nodules, so the exact number of AUC values might be a little different. However, the trend is DLSTM > LSTM > CNN. 
Thanks.

**Comment by shakjm** [comment]
> > Also, Mr. Gao, I've tried using your previous github version of train.csv and test.csv. May I know why the training and AUC stays only at 0.5 after training for so many iterations?
> 
> Hi, Shakjm, thanks for your interest. As reported in the paper, in the v2 of tumor-cifar, there is no clear difference between the pos / neg if only consider the image. So, the classical CNN / LSTM can only achieve ~0.5. As for the *.csv files, they can be generated using the data_submit.py, examples in data_submit.ipynb. Note that, there is randomness in simulating the nodules, so the exact number of AUC values might be a little different. However, the trend is DLSTM > LSTM > CNN.
> Thanks.

Hi Mr. Gao, thank you so much for your reply. I have since then generated v1 and have tried training on disrnn (new_main.py). However, the loss doesn't seem to converge (both accuracy and AUC also stays at 0.5). May I know if the codes still work for you? Could you share the pytorch version you're using?

**Comment by RiqiangGao** [comment]
> > > Also, Mr. Gao, I've tried using your previous github version of train.csv and test.csv. May I know why the training and AUC stays only at 0.5 after training for so many iterations?
> > 
> > 
> > Hi, Shakjm, thanks for your interest. As reported in the paper, in the v2 of tumor-cifar, there is no clear difference between the pos / neg if only consider the image. So, the classical CNN / LSTM can only achieve ~0.5. As for the *.csv files, they can be generated using the data_submit.py, examples in data_submit.ipynb. Note that, there is randomness in simulating the nodules, so the exact number of AUC values might be a little different. However, the trend is DLSTM > LSTM > CNN.
> > Thanks.
> 
> Hi Mr. Gao, thank you so much for your reply. I have since then generated v1 and have tried training on disrnn (new_main.py). However, the loss doesn't seem to converge (both accuracy and AUC also stays at 0.5). May I know if the codes still work for you? Could you share the pytorch version you're using?

Hi, Shakjm, I might create a typo in this simplified version. I looked back to my complete repository and updated this repository just now. Please let me know if this works for you.

**Comment by RiqiangGao** [comment]
> > > Also, Mr. Gao, I've tried using your previous github version of train.csv and test.csv. May I know why the training and AUC stays only at 0.5 after training for so many iterations?
> > 
> > 
> > Hi, Shakjm, thanks for your interest. As reported in the paper, in the v2 of tumor-cifar, there is no clear difference between the pos / neg if only consider the image. So, the classical CNN / LSTM can only achieve ~0.5. As for the *.csv files, they can be generated using the data_submit.py, examples in data_submit.ipynb. Note that, there is randomness in simulating the nodules, so the exact number of AUC values might be a little different. However, the trend is DLSTM > LSTM > CNN.
> > Thanks.
> 
> Hi Mr. Gao, thank you so much for your reply. I have since then generated v1 and have tried training on disrnn (new_main.py). However, the loss doesn't seem to converge (both accuracy and AUC also stays at 0.5). May I know if the codes still work for you? Could you share the pytorch version you're using?

Thanks for letting me know. I might have created a typo when updating this simplified repository. I looked back to my original code and just update this repository. Please let me know if this works for you. The pytorch version of mine is 0.4.

**Comment by shakjm** [comment]
> > > > Also, Mr. Gao, I've tried using your previous github version of train.csv and test.csv. May I know why the training and AUC stays only at 0.5 after training for so many iterations?
> > > 
> > > 
> > > Hi, Shakjm, thanks for your interest. As reported in the paper, in the v2 of tumor-cifar, there is no clear difference between the pos / neg if only consider the image. So, the classical CNN / LSTM can only achieve ~0.5. As for the *.csv files, they can be generated using the data_submit.py, examples in data_submit.ipynb. Note that, there is randomness in simulating the nodules, so the exact number of AUC values might be a little different. However, the trend is DLSTM > LSTM > CNN.
> > > Thanks.
> > 
> > 
> > Hi Mr. Gao, thank you so much for your reply. I have since then generated v1 and have tried training on disrnn (new_main.py). However, the loss doesn't seem to converge (both accuracy and AUC also stays at 0.5). May I know if the codes still work for you? Could you share the pytorch version you're using?
> 
> Thanks for letting me know. I might have created a typo when updating this simplified repository. I looked back to my original code and just update this repository. Please let me know if this works for you. The pytorch version of mine is 0.4.

Hi Mr. Gao, thank you so much for spending your time on looking through the codes. It is working properly now. May I know what does "add_positive" in the cifar10.yaml do? I've noticed you've changed it to True.

In your published paper, you have mentioned about testing NLST images in the model. May I know how you've done it, because there are no codes provided for them. Do you take T0, T1, T2 as 0, 1, and 2 respectively for time distance input?

Thank you again for the help. It is very much appreciated.

**Comment by RiqiangGao** [comment]
Sorry for the late. The scan in NLST has a time distance to a randomly initial date in their metafile. I would recommend digging more into the NLST. Thank you.
